aws/usageReports/ebs/dailyReport.go

```diff
@@ -0,0 +1,116 @@
+//   Copyright 2018 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package ebs
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/trackit/jsonlog"
+
+	taws "github.com/trackit/trackit-server/aws"
+	"github.com/trackit/trackit-server/aws/usageReports"
+	"github.com/trackit/trackit-server/config"
+)
+
+// fetchDailySnapshotsList sends in snapshotInfoChan the snapshots fetched from DescribeSnapshots
+// and filled by DescribeSnapshots and getSnapshotStats.
+func fetchDailySnapshotsList(ctx context.Context, creds *credentials.Credentials, awsAccount taws.AwsAccount, region string, snapshotChan chan Snapshot) error {
+	defer close(snapshotChan)
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(region),
+	}))
+	svc := ec2.New(sess)
+	snapshots, err := svc.DescribeSnapshots(&ec2.DescribeSnapshotsInput{
+		OwnerIds: []*string{
+			aws.String(awsAccount.AwsIdentity),
+		},
+	})
+	if err != nil {
+		logger.Error("Error when describing snapshots", err.Error())
+		return err
+	}
+	for _, snapshot := range snapshots.Snapshots {
+		snapshotChan <- Snapshot{
+			SnapshotBase: SnapshotBase{
+				Id:          aws.StringValue(snapshot.SnapshotId),
+				Description: aws.StringValue(snapshot.Description),
+				State:       aws.StringValue(snapshot.State),
+				Encrypted:   aws.BoolValue(snapshot.Encrypted),
+				StartTime:   aws.TimeValue(snapshot.StartTime),
+				Region:      region,
+			},
+			Tags:  getSnapshotTag(snapshot.Tags),
+			Volume: Volume{
+				Id:    aws.StringValue(snapshot.VolumeId),
+				Size:  aws.Int64Value(snapshot.VolumeSize),
```

Comment:
Make sure to run `gofmt` to have the right alignment for data in these structures

Reply:
Fixed

Reply:
Fixed